packages/starknet-snap/src/executeTxn.ts

```diff
@@ -32,16 +64,19 @@ export async function executeTxn(params: ApiParams) {
         content: panel([heading('Do you want to sign this transaction(s)?'), ...snapComponents]),
       },
     });
-
     if (!response) return false;
-
+    //In case this is the first transaction we assign a nonce of 1 to make sure it does after the deploy transaction
+    const nonceSendTransaction = accountDeployed ? undefined : 1;
```

Comment:
2 possible issues will making an error

1. when createAccount success but pending in mem pool, in that case the acc is not yet deploy, and the next txn should throw an error saying contract not found/deploy
2. when createAccount success and land to L2, however , a racing condition happening, the next txn is using an old nonce (it is a bug from starknet.js, as we already specific the nonce 1), the error should look like "nonce repreated"